store/postgres/src/deployment_store.rs

```diff
@@ -256,13 +256,24 @@ impl DeploymentStore {
         )
     }
 
-    fn check_interface_entity_uniqueness(
+    fn check_interface_entity_uniqueness<'a>(
         &self,
         conn: &PgConnection,
         layout: &Layout,
-        key: &EntityKey,
+        keys: impl Iterator<Item = &'a EntityKey> + Clone,
     ) -> Result<(), StoreError> {
+        let mut keys = keys.peekable();
+
+        // We must receive at least one key
+        let key = match keys.peek() {
+            Some(&key) => key.to_owned(),
+            None => panic!("check_interface_entity_uniqueness received no keys"),
+        };
+
         assert_eq!(&key.subgraph_id, &layout.site.deployment);
+        debug_assert!(keys
```

Comment:
All `keys` received by this function came from a previous `group_by` operation, so we can expect them to be of the same type. Nonetheless, it's always good to keep that premise under an assertion.